airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

```diff
@@ -0,0 +1,175 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import unicodedata as ud
+from re import match, sub
+
+from normalization.destination_type import DestinationType
+from normalization.transform_catalog.reserved_keywords import is_reserved_keyword
+from normalization.transform_catalog.utils import jinja_call
+
+DESTINATION_SIZE_LIMITS = {
+    DestinationType.BIGQUERY.value: 1024,
+    DestinationType.SNOWFLAKE.value: 255,
+    DestinationType.REDSHIFT.value: 127,
+    DestinationType.POSTGRES.value: 63,
+}
+
+# in DBT Versions < 19.0:
+TRUNCATE_DBT_RESERVED_SIZE = 29
+# in DBT Versions >= 19.0:
+# see https://github.com/fishtown-analytics/dbt/pull/2850
+TRUNCATE_DBT_RESERVED_SIZE_v19 = 12
+
+# We reserve this many characters from identifier names to be used for prefix/suffix for airbyte
+# before reaching the database name length limit
+TRUNCATE_RESERVED_SIZE: int = 5
+
+
+class DestinationNameTransformer:
+    """
+    Handles naming conventions in destinations for all kind of sql identifiers:
+    - schema
+    - table
+    - column
+    """
+
+    def __init__(self, integration_type: DestinationType):
+        """
+        @param integration_type is the destination type of warehouse
+        """
+        self.integration_type: DestinationType = integration_type
+
+    # Public methods
+
+    def needs_quotes(self, input_name: str) -> bool:
+        """
+        @param input_name to test if it needs to manipulated with quotes or not
+        """
+        if is_reserved_keyword(input_name, self.integration_type):
+            return True
+        if self.integration_type.value == DestinationType.BIGQUERY.value:
```

Comment:
do we need `.value` since they are both type `DestinationType`?

Reply:
I've run into issue with python enum that with the same integration type:
- `self.integration_type.value == DestinationType.BIGQUERY.value` equals true
- `self.integration_type == DestinationType.BIGQUERY` equals false